tests/cypress/e2e/hedy_page/save_button.cy.js

```diff
@@ -0,0 +1,40 @@
+import {goToHedyPage, goToHome} from "../tools/navigation/nav";
+import {loginForStudent} from "../tools/login/login";
+
+describe('Checks if save button works', () => {
+    it('Passes', () => {
+      loginForStudent();
+      cy.get('#start_programming_button').click();
+
+      // type in editor 'print test'
+      cy.get('#editor').type('\nprint test');
+      
+      cy.get('#runit').click();
+      cy.wait(3000);
+
+      // the save button:
+      cy.get('#save_program_button')
+      .should('be.visible')
+      .should('not.be.disabled')
+      .click();
+      
+      // this is for if you already have saved your level once:
+      if (cy.get('#modal-confirm')){
+        cy.get('#modal-yes-button').click();
+      }
+      
+      goToHome();
+      goToHedyPage();
+
+      // checks if it is correctly saved:
+      cy.get('#editor > .ace_scroller > .ace_content .ace_line').each((element, index) => {
+        if(index == 0) {
+          cy.get(element).should('have.text', 'print hello world!');
```

Comment:
What if we change the order or have a non-english language? Then this would fail! I would propose to first empty the editor, then add the line `print test` and then save the program. This way we are always sure the program contains one line that is the line set in this file.

Reply:
I resolved this.